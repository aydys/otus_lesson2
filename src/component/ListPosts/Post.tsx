import React from "react";

interface PostProps {
  post: {
    title: string,
    body: string,
    id: number
  },
  onClick: (id: number) => void
}

export class Post extends React.Component  <PostProps, {}> {
  constructor(props: PostProps) {
    super(props)
    this.state = {
    }
  }
  
  render() {
    const { post, onClick } = this.props;
    return <div>
      <h4>{post.title}</h4>
      <button onClick={() => onClick(post.id)}>Load</button>
    </div>
  }
}
