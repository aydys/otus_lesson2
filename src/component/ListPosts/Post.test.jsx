import React from 'react'
import { mount, render } from 'enzyme';
import { Post } from './Post'

describe('Post render', () => {
  test('render when status of post hidden', () => {
    const wrapper = render(<Post />)
    expect(wrapper).toMatchSnapshot()
  })
})