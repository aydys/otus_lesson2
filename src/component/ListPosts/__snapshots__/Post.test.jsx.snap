// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post render render when status of post hidden 1`] = `
initialize {
  "0": Node {
    "attribs": Object {},
    "children": Array [
      Node {
        "data": "Stub",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
